yt_dlp/extractor/hidive.py

```diff
@@ -60,24 +76,22 @@ def _call_api(self, video_id, title, key, data={}, **kwargs):
             'https://www.hidive.com/play/settings', video_id,
             data=urlencode_postdata(data), **kwargs) or {}
 
-    def _extract_subtitles_from_rendition(self, rendition, subtitles, parsed_urls):
-        for cc_file in rendition.get('ccFiles', []):
-            cc_url = url_or_none(try_get(cc_file, lambda x: x[2]))
-            # name is used since we cant distinguish subs with same language code
-            cc_lang = try_get(cc_file, (lambda x: x[1].replace(' ', '-').lower(), lambda x: x[0]), str)
-            if cc_url not in parsed_urls and cc_lang:
-                parsed_urls.add(cc_url)
-                subtitles.setdefault(cc_lang, []).append({'url': cc_url})
-
     def _get_subtitles(self, url, video_id, title, key, parsed_urls):
-        webpage = self._download_webpage(url, video_id, fatal=False) or ''
         subtitles = {}
-        for caption in set(re.findall(r'data-captions=\"([^\"]+)\"', webpage)):
-            renditions = self._call_api(
-                video_id, title, key, {'Captions': caption}, fatal=False,
-                note=f'Downloading {caption} subtitle information').get('renditions') or {}
-            for rendition_id, rendition in renditions.items():
-                self._extract_subtitles_from_rendition(rendition, subtitles, parsed_urls)
+        webpage = self._download_webpage(url, video_id, fatal=False) or ''
+
+        settings = self._call_api(video_id, title, key)
+        for rendition_id, rendition in settings['renditions'].items():
+            audio, version, extra = rendition_id.split('_')
+
+            for cc_file in rendition.get('ccFiles', []):
+                cc_url = url_or_none(try_get(cc_file, lambda x: x[2]))
+                # name is used since we cant distinguish subs with same language code
```

Comment:
pls explain

Reply:
Thanks for the encouragement to look into that!
I improved the comments in the new commit I made.

Reply:
I was asking to explain it to me - coz I don't understand this code/comment. You shouldn't use long name as language code. subtitles have a separate `name` field

Reply:
With it as-is, here's what --list-subs shows:
```
Language     Formats
english-caps vtt
english-subs vtt
```

I have removed that comment in the latest version of this file -- it didn't make sense to me, either. Instead, I listed out example values in that data structure.

Do you still want me to make any changes here? This comment is on an outdated version.